external/dataclass-ser/dataclass_serialization/yaml.py

```diff
@@ -0,0 +1,170 @@
+from typing import Type
+import yaml
+import dacite
+import dataclasses
+import dacite.types
+
+
+def _make_repr(
+    dumper: Type[yaml.SafeDumper],
+    cls,
+    tag=yaml.resolver.BaseResolver.DEFAULT_MAPPING_TAG,
+):
+    names = [field.name for field in dataclasses.fields(cls)]
+
+    def _represent(dumper: yaml.SafeDumper, data: cls):
+        mapping = {}
+        for name in names:
+            mapping[name] = getattr(data, name)
+
+        return dumper.represent_mapping(tag, mapping)
+
+    dumper.add_representer(cls, _represent)
+
+
+class Visitor:
+    def visit_dataclass(self, cls):
+        if cls in self.visited:
+            return
+
+        _make_repr(self.dumper, cls)
+        self.visited.add(cls)
+
+    def visit_leaf(self, cls):
+        pass
+
+    def visit_union(self, cls):
+        pass
+
+
+class DumperVisitor(Visitor):
+    def __init__(self, dumper):
+        self.visited = set()
+        self.dumper = dumper
+
+    def visit_dataclass(self, node: "DataclassNode"):
+        if node.cls in self.visited:
+            return
+
+        tag = (
+            yaml.resolver.BaseResolver.DEFAULT_MAPPING_TAG
+            if node.name
+            else "!" + node.cls.__name__
```

Comment:
how does this work if there are classes defined in different modules with the same name? e.g. `Config` defined all over the place.

Reply:
they would conflict ;)

Reply:
+1 for redundancy in module and class names haha

Reply:
yah. this is more just a proof of concept atm. It's probably worth defining a backward-compatible API namespace for our public configs regardless. Whether that namespace is flat or has some nesting is up to us.